backend/app/adapter/db/user_changelog.go

```diff
@@ -0,0 +1,89 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/short-d/short/app/adapter/db/table"
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/repository"
+)
+
+var _ repository.UserChangeLog = (*UserChangeLogSQL)(nil)
+
+// UserChangeLogSQL accesses UserChangeLog information in user_changelog table through SQL.
+type UserChangeLogSQL struct {
+	db *sql.DB
+}
+
+// GetLastViewedAt finds LastViewedAt time for given user.
+func (u UserChangeLogSQL) GetLastViewedAt(user entity.User) (time.Time, error) {
+	statement := fmt.Sprintf(`
+SELECT "%s" 
+FROM "%s"
+WHERE "%s"=$1;`,
+		table.UserChangeLog.ColumnLastViewedAt,
+		table.UserChangeLog.TableName,
+		table.UserChangeLog.ColumnEmail,
+	)
+
+	row := u.db.QueryRow(statement, user.Email)
+	lastViewedAt := time.Time{}
+	err := row.Scan(&lastViewedAt)
+	return lastViewedAt, err
+}
+
+// UpdateLastViewedAt updates LastViewedAt time for given user to currentTime.
+func (u UserChangeLogSQL) UpdateLastViewedAt(user entity.User, currentTime time.Time) (time.Time, error) {
+	lastViewedAt, err := u.GetLastViewedAt(user)
+	if err != nil {
+		return lastViewedAt, err
+	}
+
+	statement := fmt.Sprintf(`
+UPDATE "%s"
+SET %s=$1
+WHERE %s=$2
+`,
+		table.UserChangeLog.TableName,
+		table.UserChangeLog.ColumnLastViewedAt,
+		table.UserChangeLog.ColumnEmail,
+	)
+
+	_, err = u.db.Exec(
+		statement,
+		currentTime,
+		user.Email,
+	)
+	return currentTime, err
+}
+
+// CreateRelation inserts a new entry into user_changelog table.
+func (u UserChangeLogSQL) CreateRelation(user entity.User, currentTime time.Time) (time.Time, error) {
```

Comment:
Why should we return `time.Time` object?

Reply:
@alldroll I think it's because the method was originally called `CreateLastViewAt`. Yes, this need to be resolved.

Reply:
@alldroll We should only return error now.